src/rgw/rgw_op.cc

```diff
@@ -400,6 +401,10 @@ static int read_obj_policy(const DoutPrefixProvider *dpp,
   int ret = get_obj_policy_from_attr(dpp, s->cct, store, *obj_ctx,
                                      bucket_info, bucket_attrs, acl, storage_class, object, s->yield);
   if (ret == -ENOENT) {
+    if (copy_src && copy_src_exist) {
+      *copy_src_exist = false;
+      return ret;
+    }
```

Comment:
the code just below goes to great lengths to decide whether we should return ENOENT or EACCES, and that logic should apply to copy_src as well

a user without permission to list the bucket should not be able to tell whether or not the source object exists